pennylane/qchem/openfermion_obs.py

```diff
@@ -917,6 +892,124 @@ def molecular_hamiltonian(
     + (0.176276408043196) [Z2 Z3]
     """
 
+    if len(args) != 0:
+        return _molecular_hamiltonian_dispatch(*args, **kwargs)
+
+    if "symbols" in kwargs:
+        symbols = kwargs["symbols"]
+        kwargs.pop("symbols")
+        return _molecular_hamiltonian_dispatch(symbols, **kwargs)
+
+    if "molecule" in kwargs:
+        molecule = kwargs["molecule"]
+        kwargs.pop("molecule")
+        return _molecular_hamiltonian_dispatch(molecule, **kwargs)
+
+    raise NotImplementedError("Unsupported type")
+
+
+@singledispatch
+def _molecular_hamiltonian_dispatch(*args, **kwargs):
+    r"""Generate the qubit Hamiltonian of a molecule."""
+    raise NotImplementedError("Unsupported type")
```

Comment:
Is it possible to make a more informative error that tells the user "Please ensure there is a molecule/symbols/geometry passed"?

Reply:
I am going to just mark this resolved since @obliviateandsurrender has the same comment.